### PR TITLE
S3C-4368 "S3 API capabilities update"

### PR DESCRIPTION
**Pull request by clemenceauriacombe**

### CODE REVIEW

File: src/main/resources/s3capabilities.json [hunk 1]
```diff
@@ -1,284 +1,71 @@
...
+     "put_object_tagging": {},
+     "abort_multipart_upload": {},
+     "complete_multipart_upload": {},
+     "initiate_multipart_upload": {},
+     "list_parts": {},
+     "upload_part": {},
+     "upload_part_copy": {}
   }
-}
+  }
```
Comment (anurag4DSB) [inline src/main/resources/s3capabilities.json:64]:
```suggestion
}
```

A space is not needed for the last `}`

File: src/main/resources/s3capabilities.json [hunk 2]
```diff
@@ -1,284 +1,71 @@
...
+   
+   
+   
+   
+   
+  
+
```
Comment (anurag4DSB) [inline src/main/resources/s3capabilities.json:71]: Only 1 new line is needed at end of file

**Review by anurag4DSB — COMMENTED** [review]
Just some minor changes, otherwise LGTM.

File: src/main/resources/s3capabilities.json (hunk 2 above)
Comment (clemenceauriacombe) [reply src/main/resources/s3capabilities.json:71]: Thanks for your feedback Anurag

**Comment by clemenceauriacombe** [comment]
update syntax and indentation

**Review by anurag4DSB — CHANGES_REQUESTED** [review]
The indentation of the API names is 5 spaces.
It should be 4 spaces.
@clemenceauriacombe 
I can add a commit on this PR, let me know.

File: osis-core/src/test/resources/s3capabilities.json [hunk 3]
```diff
@@ -1,284 +1,64 @@
...
+     "put_object_acl": {},
+     "put_object_legal_hold": {},
+     "put_object_retention": {},
+     "put_object_tagging": {},
+     "abort_multipart_upload": {},
+     "complete_multipart_upload": {},
+     "initiate_multipart_upload": {},
+     "list_parts": {},
+     "upload_part": {},
+     "upload_part_copy": {}
```
Comment (anurag4DSB) [inline osis-core/src/test/resources/s3capabilities.json:62]:
```suggestion
    "delete_bucket": {},
    "delete_bucket_cors": {},
    "delete_bucket_encryption": {},
    "delete_bucket_lifecycle": {},
    "delete_bucket_metrics_configuration": {},
    "delete_bucket_policy": {},
    "delete_bucket_replication": {},
    "delete_bucket_website": {},
    "get_bucket_list_objects_version1": {},
    "get_bucket_list_objects_version2": {},
    "get_bucket_acl": {},
    "get_bucket_cors": {},
    "get_bucket_encryption": {},
    "get_bucket_lifecycle": {},
    "get_bucket_location": {},
    "get_bucket_metrics_configuration": {},
    "get_bucket_notification": {},
    "get_object_lock_configuration": {},
    "get_bucket_object_versions": {},
    "get_bucket_policy": {},
    "get_bucket_replication": {},
    "get_bucket_versioning": {},
    "get_bucket_website": {},
    "head_bucket": {},
    "list_bucket_metrics_configurations": {},
    "list_multipart_uploads": {},
    "put_bucket": {},
    "put_bucket_acl": {},
    "put_bucket_analytics_configuration": {},
    "put_bucket_cors": {},
    "put_bucket_encryption": {},
    "put_bucket_lifecycle": {},
    "put_bucket_metrics_configuration": {},
    "put_bucket_notification": {},
    "put_object_lock_configuration": {},
    "put_bucket_policy": {},
    "put_bucket_replication": {},
    "put_bucket_versioning": {},
    "put_bucket_website": {},
    "delete_objects": {},
    "delete_object": {},
    "delete_object_tagging": {},
    "get_object": {},
    "get_object_acl": {},
    "get_object_legal_hold": {},
    "get_object_retention": {},
    "get_object_tagging": {},
    "head_object": {},
    "put_object": {},
    "put_object_copy": {},
    "put_object_acl": {},
    "put_object_legal_hold": {},
    "put_object_retention": {},
    "put_object_tagging": {},
    "abort_multipart_upload": {},
    "complete_multipart_upload": {},
    "initiate_multipart_upload": {},
    "list_parts": {},
    "upload_part": {},
    "upload_part_copy": {}
```

Comment (anurag4DSB) [inline osis-core/src/test/resources/s3capabilities.json:62]:
```suggestion
    "delete_bucket": {},
    "delete_bucket_cors": {},
    "delete_bucket_encryption": {},
    "delete_bucket_lifecycle": {},
    "delete_bucket_metrics_configuration": {},
    "delete_bucket_policy": {},
    "delete_bucket_replication": {},
    "delete_bucket_website": {},
    "get_bucket_list_objects_version1": {},
    "get_bucket_list_objects_version2": {},
    "get_bucket_acl": {},
    "get_bucket_cors": {},
    "get_bucket_encryption": {},
    "get_bucket_lifecycle": {},
    "get_bucket_location": {},
    "get_bucket_metrics_configuration": {},
    "get_bucket_notification": {},
    "get_object_lock_configuration": {},
    "get_bucket_object_versions": {},
    "get_bucket_policy": {},
    "get_bucket_replication": {},
    "get_bucket_versioning": {},
    "get_bucket_website": {},
    "head_bucket": {},
    "list_bucket_metrics_configurations": {},
    "list_multipart_uploads": {},
    "put_bucket": {},
    "put_bucket_acl": {},
    "put_bucket_analytics_configuration": {},
    "put_bucket_cors": {},
    "put_bucket_encryption": {},
    "put_bucket_lifecycle": {},
    "put_bucket_metrics_configuration": {},
    "put_bucket_notification": {},
    "put_object_lock_configuration": {},
    "put_bucket_policy": {},
    "put_bucket_replication": {},
    "put_bucket_versioning": {},
    "put_bucket_website": {},
    "delete_objects": {},
    "delete_object": {},
    "delete_object_tagging": {},
    "get_object": {},
    "get_object_acl": {},
    "get_object_legal_hold": {},
    "get_object_retention": {},
    "get_object_tagging": {},
    "head_object": {},
    "put_object": {},
    "put_object_copy": {},
    "put_object_acl": {},
    "put_object_legal_hold": {},
    "put_object_retention": {},
    "put_object_tagging": {},
    "abort_multipart_upload": {},
    "complete_multipart_upload": {},
    "initiate_multipart_upload": {},
    "list_parts": {},
    "upload_part": {},
    "upload_part_copy": {}
```

File: src/main/resources/s3capabilities.json [hunk 4]
```diff
@@ -1,284 +1,64 @@
...
+     "put_object_acl": {},
+     "put_object_legal_hold": {},
+     "put_object_retention": {},
+     "put_object_tagging": {},
+     "abort_multipart_upload": {},
+     "complete_multipart_upload": {},
+     "initiate_multipart_upload": {},
+     "list_parts": {},
+     "upload_part": {},
+     "upload_part_copy": {}
```
Comment (anurag4DSB) [inline src/main/resources/s3capabilities.json:62]:
```suggestion
    "delete_bucket": {},
    "delete_bucket_cors": {},
    "delete_bucket_encryption": {},
    "delete_bucket_lifecycle": {},
    "delete_bucket_metrics_configuration": {},
    "delete_bucket_policy": {},
    "delete_bucket_replication": {},
    "delete_bucket_website": {},
    "get_bucket_list_objects_version1": {},
    "get_bucket_list_objects_version2": {},
    "get_bucket_acl": {},
    "get_bucket_cors": {},
    "get_bucket_encryption": {},
    "get_bucket_lifecycle": {},
    "get_bucket_location": {},
    "get_bucket_metrics_configuration": {},
    "get_bucket_notification": {},
    "get_object_lock_configuration": {},
    "get_bucket_object_versions": {},
    "get_bucket_policy": {},
    "get_bucket_replication": {},
    "get_bucket_versioning": {},
    "get_bucket_website": {},
    "head_bucket": {},
    "list_bucket_metrics_configurations": {},
    "list_multipart_uploads": {},
    "put_bucket": {},
    "put_bucket_acl": {},
    "put_bucket_analytics_configuration": {},
    "put_bucket_cors": {},
    "put_bucket_encryption": {},
    "put_bucket_lifecycle": {},
    "put_bucket_metrics_configuration": {},
    "put_bucket_notification": {},
    "put_object_lock_configuration": {},
    "put_bucket_policy": {},
    "put_bucket_replication": {},
    "put_bucket_versioning": {},
    "put_bucket_website": {},
    "delete_objects": {},
    "delete_object": {},
    "delete_object_tagging": {},
    "get_object": {},
    "get_object_acl": {},
    "get_object_legal_hold": {},
    "get_object_retention": {},
    "get_object_tagging": {},
    "head_object": {},
    "put_object": {},
    "put_object_copy": {},
    "put_object_acl": {},
    "put_object_legal_hold": {},
    "put_object_retention": {},
    "put_object_tagging": {},
    "abort_multipart_upload": {},
    "complete_multipart_upload": {},
    "initiate_multipart_upload": {},
    "list_parts": {},
    "upload_part": {},
    "upload_part_copy": {}
```

Comment (anurag4DSB) [inline src/main/resources/s3capabilities.json:62]:
```suggestion
    "delete_bucket": {},
    "delete_bucket_cors": {},
    "delete_bucket_encryption": {},
    "delete_bucket_lifecycle": {},
    "delete_bucket_metrics_configuration": {},
    "delete_bucket_policy": {},
    "delete_bucket_replication": {},
    "delete_bucket_website": {},
    "get_bucket_list_objects_version1": {},
    "get_bucket_list_objects_version2": {},
    "get_bucket_acl": {},
    "get_bucket_cors": {},
    "get_bucket_encryption": {},
    "get_bucket_lifecycle": {},
    "get_bucket_location": {},
    "get_bucket_metrics_configuration": {},
    "get_bucket_notification": {},
    "get_object_lock_configuration": {},
    "get_bucket_object_versions": {},
    "get_bucket_policy": {},
    "get_bucket_replication": {},
    "get_bucket_versioning": {},
    "get_bucket_website": {},
    "head_bucket": {},
    "list_bucket_metrics_configurations": {},
    "list_multipart_uploads": {},
    "put_bucket": {},
    "put_bucket_acl": {},
    "put_bucket_analytics_configuration": {},
    "put_bucket_cors": {},
    "put_bucket_encryption": {},
    "put_bucket_lifecycle": {},
    "put_bucket_metrics_configuration": {},
    "put_bucket_notification": {},
    "put_object_lock_configuration": {},
    "put_bucket_policy": {},
    "put_bucket_replication": {},
    "put_bucket_versioning": {},
    "put_bucket_website": {},
    "delete_objects": {},
    "delete_object": {},
    "delete_object_tagging": {},
    "get_object": {},
    "get_object_acl": {},
    "get_object_legal_hold": {},
    "get_object_retention": {},
    "get_object_tagging": {},
    "head_object": {},
    "put_object": {},
    "put_object_copy": {},
    "put_object_acl": {},
    "put_object_legal_hold": {},
    "put_object_retention": {},
    "put_object_tagging": {},
    "abort_multipart_upload": {},
    "complete_multipart_upload": {},
    "initiate_multipart_upload": {},
    "list_parts": {},
    "upload_part": {},
    "upload_part_copy": {}
```

**Review by clemenceauriacombe — COMMENTED** [review]
/approve

**Review by anurag4DSB — APPROVED** [review]

**Pull request merged**